lib/modules/apostrophe-schemas/index.js

```diff
@@ -584,6 +584,49 @@ module.exports = {
       }
     };
 
+    // Determine whether the given field is read-only
+    // based on readOnlyFields options of all fields
+
+    self.isReadOnly = function(schema, object, name) {
```

Comment:
ℹ️ logic replicated from `self.isVisible` above

Reply:
Can this be refactored so that both methods are short wrappers that pass a few property names to a shared method that implements the logic for both?

Reply:
This is a little more complicated than having just different options name `showFields` and `readOnlyFields`.
The conditions are different, making not just that easy to refactor.

Still doable, but I didn't want to spend more time on this and over-engineer it, because it's A2.
But if you think that should be refactored nonetheless, I can do it (but it'll impact showFields).

I'll give it a try

Reply:
@boutell I have factored out the code in `self.doesMatchConditionalChoices` for this file only first, just to see if it makes sense to you (if we're okay with that, I'll do the same with the frontend file).

To me, I find it a bit messy, to gain 11 lines of code in total, and I'm not happy with the way I handle the reverse condition to cope with the fact that `showFields` and `readOnlyFields` use a different logic (include vs. exclude).

We're also impacting `self.isVisible` implementation so we risk to break `showFields`, that'll need to be fully re-tested.

What do you think, is it worth it?